zio/tzngio/type-parser.go

```diff
@@ -0,0 +1,339 @@
+package tzngio
```

Comment:
NIt: We don't usually use `-` or `_` in Go source file names, so call this `typeparser.go`.